backend/app.js

```diff
@@ -1,135 +1,163 @@
+import express from 'express';
+import cors from 'cors';
+import prisma from './db/db.js'
+import bcrypt from 'bcrypt'
+import 'dotenv/config';
+import session from 'express-session';
+import { PrismaSessionStore } from '@quixo3/prisma-session-store';
+import checkAuth from './middleware/checkAuth.js';
+
+const app = express();
+
+app.use(cors({
+    origin: 'http://localhost:3000',
+    credentials: true
+}));
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.use(session({
+    secret: process.env.SESSION_SECRET,
+    resave: false,
+    saveUninitialized: false,
+    cookie: { 
+        secure: true,
+        maxAge: 7 * 24 * 60 * 60 * 1000,  //ms -> 7 days
+    },
+    store: new PrismaSessionStore(
+        prisma, {
+            checkPeriod: 60 * 1000, //ms -> 60 seconds
+            }
+        )
+}));
+
+app.use((req, res, next) => {
+    console.log(`${req.method}:${req.url}`);
+    next();
+});
+
+app.get("/", (req, res) => {
+    res.send("Hello World!");
+});
+
+app.post('/signup', async (req, res) => {
+    const {email, username, password, confirmPassword} = req.body
+
+    if (password !== confirmPassword) {
+        return res.send('Passwords do not match!');
+    }
+    
+    const existingUser = await prisma.user.findFirst({
+        where: {
+            email: email
+        }
+    })
+
+    if(existingUser){
+        return res.status(400).send('User already exists.')
+    }
+
+    const saltRounds = 10
+    const hashedPassword = bcrypt.hashSync(password, saltRounds)
+
+    const newUser = await prisma.user.create({
+        data: {
+            email: email,
+            username: username,
+            password: hashedPassword
+        }
+    })
+
+    res.send('New user registered!')
+});
+
+// TODO T32: log in 
+app.post('/login', (req, res) => {
+    res.status(200).send("Login endpoint under construction.");
+});
+
+app.post('/changeUsername', checkAuth, (req, res) => {
+    const newUsername = req.body.newUsername
+    res.send('Changed your username to: ' + newUsername);
+})
+
+// TODO T32: profile 
+app.post('/profile', checkAuth, (req, res) => {
+    res.status(200).send("Profile endpoint under construction.");
+});
+
+// T57: Profile View - Create GET route to return a specific user profile and their skills
+app.get('/profile/:userid', async (req, res) => {
+    try{
+        const {userid} = req.params
+        
+        const userProfile = await prisma.profile.findUnique({
+            where: {
+                id: userid,
+            },
+            include: {
+                profile: {
+                    include: {
+                        skills: true,
+                    },
+                }
+            },
+        });
+
+        if(!userProfile){
+            return res.status(400).send('Profile not found.')
+        }
+
+        res.send('Profile found!')
+    } catch (error) {
+        console.error("Error fetching user profile and their skills")

```

Comment:
```suggestion
        console.error("Error fetching user profile and their skills:", error)
         
```

you should also add a return response in case of an error here

Reply:
Additionally to make it more robust, we can catch specific prisma errors

https://www.prisma.io/docs/orm/reference/error-reference

```suggestion
        console.error("Error fetching user profile and their skills")
         if (error instanceof prisma.PrismaClientKnownRequestError) {
         console.error('Prisma Known Request Error:', error.message);
         return res.status(400).json({ message: 'Database error occurred.' });
       } else if (error instanceof prisma.PrismaClientUnknownRequestError) {
        console.error('Prisma Unknown Request Error:', error.message);
        } else {
         console.error('Unexpected Error:', error.message);
         return res.status(500).json({ message: 'An unexpected error occurred while 
         fetching the profile.' });
        }
```